eng/common/scripts/Get-PullRequestUrl.ps1

```diff
@@ -0,0 +1,19 @@
+param(
+  [string]$RepositoryName,
+  [string]$SourceBranch,
+  [string]$Variable,
+  [switch]$IsOutput
+)
+
+$PRUrl = $SourceBranch
+if ($SourceBranch -match "refs/pull/(\d+)/(head|merge)") {
+    $PRUrl = "https://github.com/$RepositoryName/pull/$($Matches[1])"
+}
```

Comment:
What happens if this isn't a PR?

Reply:
The purpose of this function is really to return a description of the build source for commit and PR comments. 

If we're not building from a PR, we end up with a comment like `Triggered from refs/heads/users/pahallis/cookiecutter`

I can clean this function up and name it better.   Maybe `Get-BuildSourceDescription` and have it return text based on the ref type:

**refs/pull/1234/*:** `Pull request: {repo url}/pull/1234`
**refs/tags/something:** `Tag: [something]({repo url}/tree/refs/tags/something)`
**refs/heads/something:** `Branch: [something]({repo url}/tree/refs/heads/something)`
**else:** `[{original ref}]({repo url}/tree/{original ref})`

Commits will just be a commit sha, I'm unaware of any other ref type that can be built.

Reply:
I refactored this into Get-BuildSourceDescription.ps1